reark/src/main/java/io/reark/reark/data/store/SingleItemContentProviderStore.java

```diff
@@ -5,41 +5,41 @@
 import android.net.Uri;
 import android.support.annotation.NonNull;
 
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ConcurrentMap;
-
 import io.reark.reark.utils.Log;
 import io.reark.reark.utils.Preconditions;
 import rx.Observable;
 import rx.android.schedulers.AndroidSchedulers;
 import rx.subjects.PublishSubject;
-import rx.subjects.Subject;
+
+import static rx.Observable.concat;
 
 /**
  * Created by ttuo on 26/04/15.
  */
 abstract public class SingleItemContentProviderStore<T, U> extends ContentProviderStore<T> {
+
     private static final String TAG = SingleItemContentProviderStore.class.getSimpleName();
 
-    final private ConcurrentMap<Uri, Subject<T, T>> subjectMap = new ConcurrentHashMap<>();
+    @NonNull
+    final private PublishSubject<StoreItem<T>> subjectCache = PublishSubject.create();
 
     public SingleItemContentProviderStore(@NonNull ContentResolver contentResolver) {
         super(contentResolver);
-        Preconditions.checkNotNull(contentResolver, "Content Resolver cannot be null.");
     }
 
     @NonNull
     @Override
     protected ContentObserver getContentObserver() {
-        return new ContentObserver(createHandler(this.getClass().getSimpleName())) {
+        return new ContentObserver(createHandler(getClass().getSimpleName())) {
             @Override
             public void onChange(boolean selfChange, Uri uri) {
                 super.onChange(selfChange, uri);
-                Log.v(TAG, "onChange(" + uri + ")");
 
-                if (subjectMap.containsKey(uri)) {
-                    queryOne(uri).subscribe(t -> subjectMap.get(uri).onNext(t));
-                }
+                queryOne(uri)
+                        .doOnNext(item -> Log.v(TAG, "onChange(" + uri + ')'))
```

Comment:
Not a big deal, but I'm not a fan of combining " and '. It just doesn't look consistent